From a3f92dfef413de8637513ac60eda34a83a017d36 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 22 Jul 2021 17:18:47 +0300
Subject: [PATCH] [SW-50393] gaudi: sec: blocking sw_all_rst register

This register must not be accessible for non priviledged user,
because it resets the device and might cause the host to hang.

Change-Id: I2badab61a738956e66a5eb87129a5f0f78bb7545
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 273d785..018ff17 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -791,14 +791,7 @@
 		PROT_BIT(mmPSOC_GLOBAL_CONF_TRACE_ARUSER));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x380 - 0x480 */
-	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 2);
-
-	prot_base += SNGL_REG_PROT_SIZE * 2;	/* offsets: 0x480 - 0x500 */
-	mask = ~PROT_BIT(mmPSOC_GLOBAL_CONF_SW_ALL_RST);
-	regmap_writel(mask, PRIV_PROT_REG(prot_base));
-
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x500 - 0xF80 */
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x380 - 0xF80 */
 	set_prot_bits_range_all(prot_base, PROT_SIZE_MAX - (prot_base - base));
 }
 
